libipl/p10/ipl0.C

```diff
@@ -565,8 +565,36 @@ static bool update_genesis_hwas_state(void)
 			continue;
 
 		for (const char *data : mProcChild) {
-			pdbg_for_each_target(data, proc, child)
+			pdbg_for_each_target(data, proc, child) {
+			// mark targets as non functional in the devtree
+			// if it is marked UNUSED in the MRW. This is a workaround for
+			// handling unused IOHS targets in Bonnell.	
+			if (strcmp(data , "iohs") == 0)
 			{
+				ATTR_IOHS_CONFIG_MODE_Type iohs_config;
+				if (!pdbg_target_get_attribute(child,
+						"ATTR_IOHS_CONFIG_MODE",
+						1,
+						1, &iohs_config)) {
+					ipl_log(IPL_ERROR,
+						"Attribute ATTR_IOHS_CONFIG_MODE read failed"
+						" for iohs '%s' \n",
+						pdbg_target_path(child));
+					ipl_plat_procedure_error_handler(
+						IPL_ERR_ATTR_READ_FAIL);
+					return false;
+				}
+				if (iohs_config == ENUM_ATTR_IOHS_CONFIG_MODE_UNUSED) {
```

Comment:
Missing the functional state update here for configured iohs.

if (iohs_config == ENUM_ATTR_IOHS_CONFIG_MODE_UNUSED) {
set_or_clear_state(child, false)
else
et_or_clear_state(child, target_enabled)